src/components/badge/badge.scss

```diff
@@ -5,21 +5,24 @@
  * @prop --badge-text-color: badge text color
  */
 
-
 :host {
   display: block;
 }
 
+* {
+    box-sizing: border-box;
```

Comment:
Before:
![image](https://user-images.githubusercontent.com/154725/78232775-07722a80-74d5-11ea-8c5f-f8c1dc9478d0.png)
After:
![image](https://user-images.githubusercontent.com/154725/78232795-0e993880-74d5-11ea-964b-01af74bf415d.png)

(Well, the change in vertical alignment comes from the tweaked sizes below, but mostly, setting the box-sizing did the job.)